### PR TITLE
Add tool to test application of txs to ledger

### PR DESCRIPTION
**Pull request by georgeee**
Benchmark-like tool that creates a set number of 9-account-updates transactions and applies them to a given ledger.

Explain how you tested your changes:
* [x] Ran a tool against mainnet-like ledger and measure performance

How to use the tool:

```bash
MINA_PRIVKEY_PASS= mina ledger test-apply --ledger-path ledger --privkey-path key --num-txs 128
```

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? None

### CODE REVIEW

**Comment by georgeee** [comment]
TODO: replace custom rocksdb timings with `O1trace`

File: src/app/cli/src/init/test_ledger_application.ml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+let read_privkey privkey_path =
+  let password =
+    lazy (Secrets.Keypair.Terminal_stdin.prompt_password "Enter password: ")
+  in
+  match%map Secrets.Keypair.read ~privkey_path ~password with
+  | Ok keypair ->
+      keypair
+  | Error err ->
+      eprintf "Could not read the specified keypair: %s\n"
```
Comment (emberian) [inline src/app/cli/src/init/test_ledger_application.ml:21]: Maybe use the `logger` above for all these `eprintf`? what do you think?

**Review by emberian — APPROVED** [review]
Looks good. I don't know where to get a serialized block myself. Should we have this run in CI (with small parameters)?

**Comment by georgeee** [comment]
@emberian I was using block serializations we get from the uptime system, but probably there are less exotic ways to get it.

I definitely want to come up with a more friendly flow when start working on #14800

**Comment by georgeee** [comment]
How to run the test:

**Comment by georgeee** [comment]
!ci-build-me

**Review by dkijania — APPROVED** [review]

**Pull request merged**